src/PKSim.Core/Commands/SetExpressionLocalizationCommand.cs

```diff
@@ -0,0 +1,187 @@
+         var setLocationCommand = new SetExpressionLocalizationInProteinCommand(_protein, _localization, _simulationSubject, context).Run(context);
+         Add(setLocationCommand);
+
+         if (_localization.Is(InBloodCells))
+            AddRange(updateBloodCellsExpressionParameters(context));
+
+         if (_localization.Is(InVascularEndothelium))
+            AddRange(updateVascularEndotheliumExpressionParameters(context));
+
+         if (_localization.Is(InTissue))
+            AddRange(updateTissueExpressionParameters(context));
+
+         _protein = null;
+         _simulationSubject = null;
+
+         //update properties from first command
+         this.UpdatePropertiesFrom(setLocationCommand);
+      }
+
+      private IEnumerable<IOSPSuiteCommand> updateTissueExpressionParameters(IExecutionContext context)
+      {
+         var command = new PKSimMacroCommand();
+         var parameterTask = context.Resolve<IParameterTask>();
+         //We need to iterate over all parameters defined in the protein and update the values as expected
+         var allInterstitialFractionParameters = _simulationSubject.Individual.GetAllChildren<IParameter>(x =>
+            x.IsNamed(CoreConstants.Parameters.FRACTION_EXPRESSED_INTERSTITIAL) && x.ParentContainer.IsNamed(_protein.Name));
+
+         var allTissueRelExpParameters = _simulationSubject.Individual.GetAllChildren<IParameter>(x =>
+            x.IsNamed(CoreConstants.Parameters.REL_EXP) && x.ParentContainer.IsNamed(_protein.Name));
+
+
+         command.AddRange(setParametersForFlags(parameterTask, None, InTissue, allTissueRelExpParameters.Select(x => (x, 0.0)).ToArray()));
+         command.AddRange(setParametersForFlags(parameterTask, Interstitial, Intracellular, allInterstitialFractionParameters.Select(x => (x, 1.0)).ToArray()));
+         command.AddRange(setParametersForFlags(parameterTask, Intracellular, Interstitial, allInterstitialFractionParameters.Select(x => (x, 0.0)).ToArray()));
+
+
+         return command.All();
+      }
+
+      private IEnumerable<IOSPSuiteCommand> updateVascularEndotheliumExpressionParameters(IExecutionContext context)
+      {
+         var command = new PKSimMacroCommand();
+         var parameterTask = context.Resolve<IParameterTask>();
+         var f_exp_vasc_apical = _protein.Parameter(CoreConstants.Parameters.FRACTION_EXPRESSED_VASC_ENDO_APICAL);
+         var f_exp_vasc_endosome = _protein.Parameter(CoreConstants.Parameters.FRACTION_EXPRESSED_VASC_ENDO_ENDOSOME);
+         var rel_exp_vasc = _protein.Parameter(CoreConstants.Parameters.REL_EXP_VASC_ENDO);
+
+         //not in vasc endo => set expression to 0
+         command.AddRange(setParametersForFlags(parameterTask, None, InVascularEndothelium, (rel_exp_vasc, 0)));
+         command.AddRange(setParametersForFlags(parameterTask, VascMembraneApical | VascEndosome, VascMembraneBasolateral, (f_exp_vasc_apical, 0.5),
+            (f_exp_vasc_endosome, 0.5)));
```

Comment:
should we name the constants VascMembranePlasma and VascMembraneTissue instead of VascMembraneApical and VascMembraneBasolateral? Maybe it's only me but I can never rememeber which is which with Apical/Basolateral

Reply:
That's how they were written. But all parameters are Apical and Basolateral etc.. and it makes it very hard to see if stuff are consistent. Apical should always go with Apical and I was struggling because I had named the flags Plasma and Tissue.
So I would suggest to not rename
And I agree this is not clear for me either :)(